tests/test_connector.rs

```diff
@@ -1151,3 +1151,60 @@ fn assert_execution_status_failure(
         _ => panic!("{}", panic_msg),
     }
 }
+
+#[test]
+fn test_ft_transfer_max_value() {
+    let (_, contract) = init(CUSTODIAN_ADDRESS);
+    call_deposit_near(&contract, CONTRACT_ACC);
+
+    let transfer_amount = u128::MAX;
+    let res = contract.call(
+        CONTRACT_ACC.to_string(),
+        "ft_transfer",
+        json!({
+            "receiver_id": DEPOSITED_RECIPIENT,
+            "amount": transfer_amount.to_string(),
+            "memo": "transfer memo"
+        })
+        .to_string()
+        .as_bytes(),
+        DEFAULT_GAS,
+        1,
+    );
+    let promises = res.promise_results();
+    let promise = &promises[promises.len() - 3];
+    eprintln!("{:#?}", promise.as_ref().unwrap().outcome().clone().status);
+    assert_execution_status_failure(
+        promise.as_ref().unwrap().outcome().clone().status,
+        "ERR_NOT_ENOUGH_BALANCE",
+        "Expected failure as the amount too large",
+    );
+}
+
+#[test]
+fn test_ft_transfer_empty_value() {
+    let (_, contract) = init(CUSTODIAN_ADDRESS);
+    call_deposit_near(&contract, CONTRACT_ACC);
+
+    let res = contract.call(
+        CONTRACT_ACC.to_string(),
+        "ft_transfer",
+        json!({
+            "receiver_id": DEPOSITED_RECIPIENT,
+            "amount": "",
+            "memo": "transfer memo"
+        })
+        .to_string()
+        .as_bytes(),
+        DEFAULT_GAS,
+        1,
+    );
+    let promises = res.promise_results();
+    let promise = &promises[promises.len() - 3];
+    eprintln!("{:#?}", promise.as_ref().unwrap().outcome().clone().status);
+    assert_execution_status_failure(
+        promise.as_ref().unwrap().outcome().clone().status,
+        "ERR_FAILED_PARSE_U128",
+        "Expected failure as empty string can't be parsed to u128",
+    );
```

Comment:
As this just checks the string->u128 conversion, I think it's better to have this in the unit test in `src/parameters.rs`. The integration test for this case seems to be redundant.

Reply:
But that test covers empty parameter as well.

Reply:
The empty `amount` parameter will fail in the unit test as well.

Reply:
But we can test all flow with all arguments. And that part of the integration tests.

Reply:
I think we could have both integration tests and unit test. I think it is generally the case that it is not redundant to have both because they do subtly different things. Unit test ensure a components works correctly in isolation, while integration tests ensure a feature (which may interact with multiple components) works correctly.

The additional advantage of unit tests in this particular case is that they are lighter-weight so we can test more input values in less time (indeed we could even randomly generate valid and invalid strings to try and parse).